pulp_deb/tests/functional/api/test_crud_remotes.py

```diff
@@ -1,210 +1,157 @@
-    def test_02_read_remotes(self):
-        """Read a remote by its name."""
-        page = self.remote_api.list(name=self.remote.name)
-        self.assertEqual(len(page.results), 1)
-        for key, val in self.remote.to_dict().items():
-            with self.subTest(key=key):
-                self.assertEqual(page.results[0].to_dict()[key], val, key)
-
-    @skip_if(bool, "remote", False)
-    def test_03_partially_update(self):
-        """Update a remote using HTTP PATCH."""
-        body = _gen_verbose_remote()
-        response = self.remote_api.partial_update(self.remote.pulp_href, body)
-        monitor_task(response.task)
-        for key in ("username", "password"):
-            del body[key]
-        type(self).remote = self.remote_api.read(self.remote.pulp_href)
-        for key, val in body.items():
-            with self.subTest(key=key):
-                self.assertEqual(self.remote.to_dict()[key], val, key)
-
-    @skip_if(bool, "remote", False)
-    def test_04_fully_update(self):
-        """Update a remote using HTTP PUT."""
-        body = _gen_verbose_remote()
-        response = self.remote_api.update(self.remote.pulp_href, body)
-        monitor_task(response.task)
-        for key in ("username", "password"):
-            del body[key]
-        type(self).remote = self.remote_api.read(self.remote.pulp_href)
-        for key, val in body.items():
-            with self.subTest(key=key):
-                self.assertEqual(self.remote.to_dict()[key], val, key)
-
-    @skip_if(bool, "remote", False)
-    def test_05_delete(self):
-        """Delete a remote."""
-        response = self.remote_api.delete(self.remote.pulp_href)
-        monitor_task(response.task)
-        with self.assertRaises(ApiException):
-            self.remote_api.read(self.remote.pulp_href)
-
-
-class CreateRemoteNoURLTestCase(unittest.TestCase):
-    """Verify whether is possible to create a remote without a URL."""
-
-    def test_all(self):
-        """Verify whether is possible to create a remote without a URL.
-
-        This test targets the following issues:
-
-        * `Pulp #3395 <https://pulp.plan.io/issues/3395>`_
-        * `Pulp Smash #984 <https://github.com/pulp/pulp-smash/issues/984>`_
-        """
-        body = gen_deb_remote()
-        del body["url"]
-        with self.assertRaises(ApiException):
-            deb_remote_api.create(body)
-
-
-class RemoteDownloadPolicyTestCase(unittest.TestCase):
-    """Verify download policy behavior for valid and invalid values.
 
-    In Pulp 3, there are are different download policies.
+def test_create_remote_repository_with_same_name(
+    apt_remote_api, remote_repository, verbose_remote_data
+):
+    """Verify whether it is possible to create a remote repository with the same
+    name.
+    """
+    body = deepcopy(verbose_remote_data)
+    body["name"] = remote_repository.name
+    with pytest.raises(ApiException):
+        apt_remote_api.create(body)
```

Comment:
This is totally valid. We recommend that users use `gen_object_with_cleanup` which works almost the same (see example).
```suggestion
        gen_object_with_cleanup(apt_remote_api, body)
```

This is preferred because it provides server-side auto-removal at test teardown time. Even in the case where an exception is expected, in the event we regressed and it accidentally created it, the test would fail but it would get cleanup up there too.